lib/persisted-model.js

```diff
@@ -616,15 +621,36 @@ module.exports = function(registry) {
       http: { verb: 'post', path: '/' },
     });
 
-    setRemoting(PersistedModel, 'upsert', {
+    var upsertOptions = {
       aliases: ['updateOrCreate'],
       description: 'Update an existing model instance or insert a new one into the data source.',
       accessType: 'WRITE',
       accepts: { arg: 'data', type: 'object', http: { source: 'body' }, description:
         'Model instance data' },
       returns: { arg: 'data', type: typeName, root: true },
-      http: { verb: 'put', path: '/' },
-    });
+      http: [{ verb: 'patch', path: '/' }],
+    };
+
+    if (!options.replaceOnPUT) {
+      upsertOptions.http.push({ verb: 'put', path: '/' });
+    }
+    setRemoting(PersistedModel, 'upsert', upsertOptions);
+
+    var replaceOrCreateOptions = {
+      description: 'Replace an existing model instance or insert a new one into the data source.',
+      accessType: 'WRITE',
+      accepts: { arg: 'data', type: 'object', http: { source: 'body' }, description:
+        'Model instance data' },
+      returns: { arg: 'data', type: typeName, root: true },
+      http: { verb: 'post', path: '/replaceOrCreate' },
+    };
+
+    if (options.replaceOnPUT) {
+      replaceOrCreateOptions.http.verb = 'put';
```

Comment:
Should we preserve `POST /replaceOrCreate` endpoint for people upgrading from LoopBack 2.x to 3.0?